File: frontend/src/app/modules/supplier/services/supplier.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface SupplierDashboardStats {
  assignedOrders: number;
  pendingDeliveries: number;
  completedDeliveries: number;
  upcomingDeliveries: number;
  notifications: any[];
}

@Injectable({
  providedIn: 'root'
})
export class SupplierService {
  private http = inject(HttpClient);
  private apiUrl = '/api/supplier';

  getDashboardStats(supplierId: number): Observable<SupplierDashboardStats> {
    return this.http.get<SupplierDashboardStats>(`${this.apiUrl}/dashboard/${supplierId}`);
  }

  getPurchaseOrders(supplierId: number): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/purchase-orders/${supplierId}`);
  }

  updateOrderStatus(orderId: number, status: string): Observable<any> {
    return this.http.put(`${this.apiUrl}/purchase-orders/${orderId}/status?status=${status}`, {});
  }

  confirmETA(etaData: any): Observable<any> {
    return this.http.post(`${this.apiUrl}/eta`, etaData);
  }

  getDeliveryHistory(supplierId: number): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/delivery-history/${supplierId}`);
  }
}

File: backend/src/main/java/com/supplysync/modules/supplier/service/SupplierService.java
package com.supplysync.modules.supplier.service;

import com.supplysync.entity.PurchaseOrder;
import com.supplysync.entity.ETAConfirmation;
import com.supplysync.modules.supplier.dto.SupplierDashboardDTO;
import java.util.List;

public interface SupplierService {
    SupplierDashboardDTO getDashboardStats(Long supplierId);
    List<PurchaseOrder> getAssignedOrders(Long supplierId);
    PurchaseOrder updateOrderStatus(Long orderId, String status);
    ETAConfirmation confirmETA(ETAConfirmation eta);
    List<ETAConfirmation> getDeliveryHistory(Long supplierId);
}
